packages/react/src/contexts/index.tsx

```diff
@@ -0,0 +1,125 @@
+import { Children, createContext, Fragment, isValidElement, PropsWithChildren, ReactNode, useContext } from "react";
+import type { RoboWizard, StepConfig } from "robo-wizard";
+import { useWizard } from "../hooks";
+
+/**
+ * @param props - accepts same props to match [[StepConfig]] for [[useWizard]], as well as the `element` to display
+ **/
+export function Step(_props: StepConfig & { element: ReactNode | null }): React.ReactElement | null {
+  return null;
+}
+
+/**
+ * @param steps = The Children opaque object passed to components, only Step components will be processed
+ **/
+export function createStepsFromChildren(steps: ReactNode): Array<StepConfig & { element: ReactNode }> {
+  return Children.map(steps, element => {
+    if (isValidElement(element) === false) return;
+    if (!element) return;
+    if (typeof element === 'object' && 'type' in element) {
+      if (element.type === Fragment) {
+        return createStepsFromChildren(element.props.children);
+      }
+      if (element.type === Step) {
+        return {
+          name: element.props.name,
+          next: element.props.next,
+          previous: element.props.previous,
+          element: element.props.element,
+        }
+      }
+    }
+    return;
+  })?.flat().filter(Boolean) ?? [];
+}
+
+/**
+ * @internal
+ **/
+export const WizardContext = createContext<null | RoboWizard>(null);
```

Comment:
exporting for potential reuse in the react-router package, which would allow the context hook to be reused.